epictrack-api/src/api/models/work.py

```diff
@@ -159,7 +159,17 @@ def filter_by_search_criteria(cls, query, search_filters: WorkplanDashboardSearc
     @classmethod
     def _filter_by_staff_id(cls, query, staff_id):
         if staff_id is not None:
-            subquery = exists().where(and_(Work.id == StaffWorkRole.work_id, StaffWorkRole.staff_id == staff_id))
+            is_active_team_member = and_(StaffWorkRole.staff_id == staff_id, StaffWorkRole.is_active)
+            is_epd_on_work = Work.responsible_epd_id == staff_id
+            subquery = exists().where(
+                and_(
```

Comment:
Could you please check StaffWorkRole.is_active in find_staff_for_works. Currently if you make a team member inactive in teams tab, he can see the workplan as we are not checking the StaffWorkRole active status.

Reply:
Done, thanks for the suggestion